scipy/stats/_sobol.pyx

```diff
@@ -138,13 +138,9 @@ cdef int low_0_bit(const int x) nogil:
         Position of the right-most 0 bit.
 
     """
-    cdef int z = x
-    cdef int i = 0
-    while True:
+    cdef int i = 1
+    while x & (1 << (i-1)) != 0:
```

Comment:
I finally took some time to get this myself... This code really makes more sense.

I have one last comment. Would it make sense to actually start at 0 so that we don't do `i-1` in the loop? 
```suggestion
    cdef int i = 0
    while x & (1 << i) != 0:
```
Then we have to do `i + 1` before returning.

Reply:
Thank you for your comment.
Yes, it makes sense! It slightly improves the performance.